contrib/internal/httptrace/httptrace_test.go

```diff
@@ -24,11 +30,124 @@ func TestStartRequestSpan(t *testing.T) {
 	s, _ := StartRequestSpan(r)
 	s.Finish()
 	spans := mt.FinishedSpans()
-
 	require.Len(t, spans, 1)
 	assert.Equal(t, "example.com", spans[0].Tag("http.host"))
 }
 
+// a test logger for TestClientIP
+type testLogger struct {
+	mu    sync.RWMutex
+	lines []string
+}
```

Comment:
NIT: Could you reuse the RecordLogger from there? It doesn't seem to have a Reset method right now, but that could probably be added.

https://github.com/DataDog/dd-trace-go/blob/74a8e95a31d9fc95d2f1a7b804fabe7f2467cb0c/internal/log/log.go#L200-L220